### PR TITLE
Issue 21396 fix

### PR DESCRIPTION
**Pull request by ahardin-rh**
Addresses https://github.com/openshift/openshift-docs/issues/21396

@dmesser PTAL

cc @adellape

### CODE REVIEW

File: modules/olm-enabling-operator-restricted-network.adoc [hunk 1]
```diff
@@ -12,21 +12,12 @@ for your Operator to run properly in a restricted network environment:
...
-kind: Deployment
-metadata:
-  name: etcd-operator
-  annotations:
-    default: quay.io/coreos/etcd@sha256:12345
-    olm.relatedImage.etcd-2.1.5: quay.io/coreos/etcd@sha256:12345
-    olm.relatedImage.etcd-3.1.1: quay.io/coreos/etcd@sha256:12345
-...
-----
+* in `spec.relatedImages`.
```
Comment (dmesser) [inline modules/olm-enabling-operator-restricted-network.adoc:18]: We need this example here:

```yaml
spec:
  relatedImages:
    - name: etcd-operator
      value: quay.io/etcd-operator/operator@sha256:d134a9865524c29fcf75bbc4469013bc38d8a15cb5f41acfddb6b9e492f556e4
    - name: etcd-image
      value: quay.io/etcd-operator/etcd@sha256:13348c15263bd8838ec1d5fc4550ede9860fcbb0f843e48cbccec07810eebb68
```

File: modules/olm-enabling-operator-restricted-network.adoc [hunk 2]
```diff
@@ -12,21 +12,12 @@ for your Operator to run properly in a restricted network environment:
...
+* in the `env` section of the Operators Deployments when declaring environment
```
Comment (dmesser) [inline modules/olm-enabling-operator-restricted-network.adoc:19]: We need this example here:

```yaml
spec:
  install:
    spec:
      deployments:
      - name: etcd-operator-v3.1.1
        spec:
          replicas: 1
          selector:
            matchLabels:
              name: etcd-operator
          strategy:
            type: Recreate
          template:
            metadata:
              labels:
                name: etcd-operator
            spec:
              containers:
              - args:
                - /opt/etcd/bin/etcd_operator_run.sh
                env:
                - name: WATCH_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.annotations['olm.targetNamespaces']
                - name: ETCD_OPERATOR_DEFAULT_ETCD_IMAGE
                  value: quay.io/etcd-operator/etcd@sha256:13348c15263bd8838ec1d5fc4550ede9860fcbb0f843e48cbccec07810eebb68
                - name: ETCD_LOG_LEVEL
                  value: INFO
                image: quay.io/etcd-operator/operator@sha256:d134a9865524c29fcf75bbc4469013bc38d8a15cb5f41acfddb6b9e492f556e4
                imagePullPolicy: IfNotPresent
                livenessProbe:
                  httpGet:
                    path: /healthy
                    port: 8080
                  initialDelaySeconds: 10
                  periodSeconds: 30
                name: etcd-operator
                readinessProbe:
                  httpGet:
                    path: /ready
                    port: 8080
                  initialDelaySeconds: 10
                  periodSeconds: 30
                resources: {}
              serviceAccountName: etcd-operator
    strategy: deployment

```

**Review by dmesser — CHANGES_REQUESTED** [review]
This looks great but we also need example code.

**Comment by ahardin-rh** [comment]
@dmesser Thank you! Updated.
@emmajiafan Can you please provide a QE review?

File: modules/olm-enabling-operator-restricted-network.adoc [hunk 3]
```diff
@@ -12,21 +12,71 @@ for your Operator to run properly in a restricted network environment:
...
-    default: quay.io/coreos/etcd@sha256:12345
-    olm.relatedImage.etcd-2.1.5: quay.io/coreos/etcd@sha256:12345
-    olm.relatedImage.etcd-3.1.1: quay.io/coreos/etcd@sha256:12345
-...
+spec:
+  relatedImages:
+    - name: etcd-operator
+      value: quay.io/etcd-operator/operator@sha256:d134a9865524c29fcf75bbc4469013bc38d8a15cb5f41acfddb6b9e492f556e4
+    - name: etcd-image
+      value: quay.io/etcd-operator/etcd@sha256:13348c15263bd8838ec1d5fc4550ede9860fcbb0f843e48cbccec07810eebb68
```
Comment (dmesser) [inline modules/olm-enabling-operator-restricted-network.adoc:26]:
```suggestion
[source,yaml]
---
spec:
  relatedImages: <1>
    - name: etcd-operator <2>
      value: quay.io/etcd-operator/operator@sha256:d134a9865524c29fcf75bbc4469013bc38d8a15cb5f41acfddb6b9e492f556e4 <3>
    - name: etcd-image
      value: quay.io/etcd-operator/etcd@sha256:13348c15263bd8838ec1d5fc4550ede9860fcbb0f843e48cbccec07810eebb68
---
<1> Create a `relatedImages` section and set the list of related images.
<2> Specify a unique identifier for the image
<3> Specify each image by a digest (SHA), not by an image tag.
```

File: modules/olm-enabling-operator-restricted-network.adoc [hunk 4]
```diff
@@ -12,21 +12,71 @@ for your Operator to run properly in a restricted network environment:
...
+                readinessProbe:
+                  httpGet:
+                    path: /ready
+                    port: 8080
+                  initialDelaySeconds: 10
+                  periodSeconds: 30
+                resources: {}
+              serviceAccountName: etcd-operator
+    strategy: deployment
+----
```
Comment (dmesser) [inline modules/olm-enabling-operator-restricted-network.adoc:79]:
```suggestion
[source,yaml]
----
spec:
  install:
    spec:
      deployments:
      - name: etcd-operator-v3.1.1
        spec:
          replicas: 1
          selector:
            matchLabels:
              name: etcd-operator
          strategy:
            type: Recreate
          template:
            metadata:
              labels:
                name: etcd-operator
            spec:
              containers:
              - args:
                - /opt/etcd/bin/etcd_operator_run.sh
                env:
                - name: WATCH_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.annotations['olm.targetNamespaces']
                - name: ETCD_OPERATOR_DEFAULT_ETCD_IMAGE <1>
                  value: quay.io/etcd-operator/etcd@sha256:13348c15263bd8838ec1d5fc4550ede9860fcbb0f843e48cbccec07810eebb68 <2>
                - name: ETCD_LOG_LEVEL
                  value: INFO
                image: quay.io/etcd-operator/operator@sha256:d134a9865524c29fcf75bbc4469013bc38d8a15cb5f41acfddb6b9e492f556e4 <3>
                imagePullPolicy: IfNotPresent
                livenessProbe:
                  httpGet:
                    path: /healthy
                    port: 8080
                  initialDelaySeconds: 10
                  periodSeconds: 30
                name: etcd-operator
                readinessProbe:
                  httpGet:
                    path: /ready
                    port: 8080
                  initialDelaySeconds: 10
                  periodSeconds: 30
                resources: {}
              serviceAccountName: etcd-operator
    strategy: deployment
----
<1> Inject the images referenced by the Operator via environment variables
<2> Specify each image by a digest (SHA), not by an image tag.
<3> Also reference the Operator container image by a digest (SHA), not by an image tag.
```

**Review by dmesser — CHANGES_REQUESTED** [review]
The existing section below

```
.Prerequisites

* An Operator project with a CSV
.Procedure
* In your Operator's CSV, define a list of any related images:
+
[source,yaml]
----
kind: ClusterServiceVersion
metadata:
  name: etcd-operator
spec:
...
  relatedImages: <1>
  - name: default
    image: quay.io/coreos/etcd@sha256:12345 <2>
  - name: etcd-2.1.5
    image: quay.io/coreos/etcd@sha256:12345 <2>
  - name: etcd-3.1.1
    image: quay.io/coreos/etcd@sha256:12345 <2>
----
<1> Create a `relatedImages` section and set the list of related images.
<2> Specify each image by a digest (SHA), not by an image tag.
```

can be removed.

**Comment by ahardin-rh** [comment]
@dmesser Thanks! The latest round of updates is complete.
@emmajiafan Can you please provide QE review?
Preview build: http://file.rdu.redhat.com/~ahardin/April-30-2020/issue-21396/operators/operator_sdk/osdk-generating-csvs.html#olm-enabling-operator-for-restricted-network_osdk-generating-csvs

**Review by adellape — APPROVED** [review]
:+1:

**Pull request merged**